qiskit/algorithms/evolvers/evolution_problem.py

```diff
@@ -44,15 +45,124 @@ def __init__(
             initial_state: Quantum state to be evolved.
             aux_operators: Optional list of auxiliary operators to be evaluated with the
                 evolved ``initial_state`` and their expectation values returned.
+            truncation_threshold: Defines a threshold under which values can be assumed to be 0.
+                Used when ``aux_operators`` is provided.
             t_param: Time parameter in case of a time-dependent Hamiltonian. This
                 free parameter must be within the ``hamiltonian``.
             hamiltonian_value_dict: If the Hamiltonian contains free parameters, this
                 dictionary maps all these parameters to values.
+
+        Raises:
+            ValueError: If non-positive time of evolution is provided.
+            ValueError: If no ``initial_state`` is provided.
+            ValueError: If not all parameter values are provided.
         """
 
+        self.t_param = t_param
+        self.hamiltonian_value_dict = hamiltonian_value_dict
         self.hamiltonian = hamiltonian
         self.time = time
         self.initial_state = initial_state
         self.aux_operators = aux_operators
-        self.t_param = t_param
-        self.hamiltonian_value_dict = hamiltonian_value_dict
+        self.truncation_threshold = truncation_threshold
+
+    @property
+    def hamiltonian(self) -> OperatorBase:
+        """Returns a hamiltonian."""
+        return self._hamiltonian
+
+    @hamiltonian.setter
+    def hamiltonian(self, hamiltonian) -> None:
+        """
+        Sets a hamiltonian and validates it.
+
+        Raises:
+            ValueError: If no Hamiltonian is provided.
+            ValueError: If Hamiltonian parameters cannot be bound with data provided.
+        """
+        self._hamiltonian = None
+        if hamiltonian is None:
+            raise ValueError(
+                "No ``hamiltonian`` provided for the EvolutionProblem. It is required."
+            )
+        # TODO SparsePauliOp does not have .parameters because it is not allowed to be parametrized.
+        #  Can we handle this better than with an if?
+        if not isinstance(hamiltonian, SparsePauliOp):
+            self._check_parameters(hamiltonian, self.hamiltonian_value_dict, self.t_param)
+
+        self._hamiltonian = hamiltonian
+
+    @property
+    def time(self) -> float:
+        """Returns time."""
+        return self._time
+
+    @time.setter
+    def time(self, time) -> None:
```

Comment:
Missing type hint - I commented in the hamiltonian setter about the other setters such as this one, initial state one, but thought it might get lost in the other text so did this too.